modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rel/IgniteMergeJoin.java

```diff
@@ -86,6 +90,7 @@ public IgniteMergeJoin(RelInput input) {
             input.getInputs().get(0),
             input.getInputs().get(1),
             input.getExpression("condition"),
+            null,
```

Comment:
Value is lost during serialization/deserialization. You should use `input.getBitSet("matchingNulls")` here and `.item("matchingNulls", matchingNulls)` in `explainTerms`